ipfx/bin/run_feature_extraction.py

```diff
@@ -34,6 +34,37 @@ def embed_spike_times(input_nwb_file, output_nwb_file, sweep_features):
     logging.debug("Embedded spike times into output.nwb file")
 
 
+def collect_spike_times(sweep_features):
+
+    spikes = {}
+
+    for sweep_num in sweep_features:
+        spike_features = sweep_features[sweep_num]['spikes']
+        spike_times = [ s['threshold_t'] for s in spike_features ]
+        spikes[sweep_num] = spike_times
+
+    return spikes
+
+def embed_spike_times(input_nwb_file, output_nwb_file, sweep_spikes):
+
+    tmp_nwb_file = output_nwb_file + ".tmp"
+
+    shutil.copy(input_nwb_file, tmp_nwb_file)
+
+    nwb_data = writer.create_nwb_writer(tmp_nwb_file)
+    for sweep_num,spike_times in sweep_spikes.items():
+        nwb_data.add_spike_times(sweep_num, spike_times)
+
+    nwb_data.save()
+
+    try:
+        shutil.move(tmp_nwb_file, output_nwb_file)
```

Comment:
Why is there a need to rename the nwbfile? Wouldn't it be better to specify a output savepath to the `NWBWriter.save()` method?

Reply:
My understanding is that decision to use an intermediate tmp file was made because nwb1 had issues with corrupting files. Thus, the input nwb file is copied to tmp, spikes were added and if all went well, the output nwb is created. If we bypass the tmp file, then output nwb would be created which may not have spikes, or if it gets corrupted then it would also mess up an output nwb from the previous pipeline run.